src/HTTP/Request/ListPayments.php

```diff
@@ -0,0 +1,299 @@
+            throw new InvalidArgumentException('Expected integer for offset; ' . gettype($offset) . ' given');
+        } elseif ($offset < 0) {
+            throw new InvalidArgumentException("Expected zero or a positive integer for offset; {$offset} given");
+        }
+
+        $this->offset = $offset;
+
+        return $this;
+    }
+
+    /**
+     * @param array $tokens
+     * @return \Afterpay\SDK\HTTP\Request\ListPayments
+     * @throws \Afterpay\SDK\Exception\InvalidArgumentException
+     */
+    public function setTokens($tokens)
+    {
+        if (! is_array($tokens)) {
+            throw new InvalidArgumentException('Expected array for tokens; ' . gettype($tokens) . ' given');
+        } elseif (array_sum(array_map('is_string', $tokens)) != count($tokens)) {
+            throw new InvalidArgumentException("Expected an array of string elements; one or more non-string elements given");
+        }
+
+        $this->tokens = $tokens;
+
+        return $this;
+    }
+
+    /**
+     * @param array $ids
+     * @return \Afterpay\SDK\HTTP\Request\ListPayments
+     * @throws \Afterpay\SDK\Exception\InvalidArgumentException
+     */
+    public function setIds($ids)
+    {
+        if (! is_array($ids)) {
+            throw new InvalidArgumentException('Expected array for ids; ' . gettype($ids) . ' given');
+        } elseif (array_sum(array_map('is_string', $ids)) != count($ids)) {
+            throw new InvalidArgumentException("Expected an array of string elements; one or more non-string elements given");
+        }
+
+        $this->ids = $ids;
+
+        return $this;
+    }
+
+    /**
+     * @param array $merchantReferences
+     * @return \Afterpay\SDK\HTTP\Request\ListPayments
+     * @throws \Afterpay\SDK\Exception\InvalidArgumentException
+     */
+    public function setMerchantReferences($merchantReferences)
+    {
+        if (! is_array($merchantReferences)) {
+            throw new InvalidArgumentException('Expected array for merchantReferences; ' . gettype($merchantReferences) . ' given');
+        } elseif (array_sum(array_map('is_string', $merchantReferences)) != count($merchantReferences)) {
+            throw new InvalidArgumentException("Expected an array of string elements; one or more non-string elements given");
+        }
+
+        $this->merchantReferences = $merchantReferences;
+
+        return $this;
+    }
+
+    /**
+     * @param array $statuses
+     * @return \Afterpay\SDK\HTTP\Request\ListPayments
+     * @throws \Afterpay\SDK\Exception\InvalidArgumentException
+     */
+    public function setStatuses($statuses)
+    {
+        if (! is_array($statuses)) {
+            throw new InvalidArgumentException('Expected array for statuses; ' . gettype($statuses) . ' given');
+        } elseif (array_sum(array_map('is_string', $statuses)) != count($statuses)) {
+            throw new InvalidArgumentException("Expected an array of string elements; one or more non-string elements given");
+        }
```

Comment:
Looks like it only accepts 'APPROVED' and 'DECLINED'? I haven't tested it though.
```suggestion
        } elseif (count(array_diff($statuses, ['APPROVED', 'DECLINED'])) > 0) {
             throw new InvalidArgumentException('Expected "APPROVED" and "DECLINED" only for statuses');
        }
```

Reply:
Does the array diff need to be case-insensitive? ie could `['approved']` be passed in to `$statuses`?
I believe array_diff is case sensitive. array_udiff could be used with a case insensitive compare as the third param depending on requirements.

Reply:
Great feedback both. @chesterz-afterpay, thanks for the suggested edit. I won't commit it from here because @ScottAntonacAP is correct, the options are case insensitive. Will push a new commit to address this, along with a corresponding integration test.

Reply:
Addressed in 091575c.